lib/datadog/appsec/waf.rb

```diff
@@ -482,6 +479,33 @@ class Handle
 
         attr_reader :ruleset_info
 
+        def self.update(handle, data)
```

Comment:
I'd rather have this be an instance method on the handle we operate on (the first argument here)

Reply:
The two possible APIs we could have are:

Instance method. Calling the API would look like this:
```ruby
handle = Hanlde.new(....)

new_handle = handle.update(data)

if new_handle
  handle.finalize
end
```

Class method. Calling the API would look like this:
```ruby
handle = Handle.new(....)

new_handle = Handle.update(handle, data)

if new_handle
  handle.finalize
end
```

I don't know which is best for us in this case. Could you explain more about your reasoning? That way I can understand better your preference

Reply:
IIUC your reasoning is that `new_handle = Handle.update(handle, data)` is akin to `Handle.new`.

My reasoning is that `new_handle = handle.update(data)` is akin to `#dup`, `#map`, `#merge`, etc... 

There is no "best" choice here, although I would argue that `#update` feels more Ruby-like to me. One could argue that `#update` sounds like it would update in place, although e.g `#map` does not while `#map!` does. So I feel like `#update` returning a new handle and not mutating the old one is fine. Maybe we should not call it `#update` but `#merge`?

In addition it's consistent with `ddwaf_update`, which to be fair is not a motivation in itself to decide about this Ruby API as we're abstracting C away, but it gives additional credence to `#update`.

Reply:
I like `merge` over `update` 😄